src/adapters/adapter.js

```diff
@@ -5,6 +5,24 @@ class Adapter {
     this.store = store;
   }
 
+  _setCacheTime(storeName, cacheKey){
+    if(!this.store.get(storeName)){
+        this.store.set(storeName, {});
+    }
+    const cachedRepos = this.store.get(storeName);
+    cachedRepos[cacheKey] = new Date().getTime();
+    this.store.set(storeName, cachedRepos);
+  }
+
+  _reloadCache(storeName, cacheKey, cb, reloadAfterDay = 29){
+    const cachedRepos = this.store.get(storeName);
+    const cachedRepoTime = cachedRepos[cacheKey];
+    const timeDiff = (cachedRepoTime - new Date().getTime()) / (1000 * 60 * 60 * 24);
+    if(timeDiff > reloadAfterDay){
+      cb()
```

Comment:
When force reload or expire, I think we should delete the cache, no? Otherwise, user cache will keep growing forever. 

Related is: if users stop accessing a repo (that was cached), do browsers clean up cache automatically? Or the cache will keep growing...

Reply:
I didn't consider the case where users stop using the repo. 
I borrowed this cache idea from PWA, from my experience, the service worker will handle the cache.

In this case, to be honest, I don't know how the browsers will handle it. Reading from [this](https://developers.google.com/web/fundamentals/instant-and-offline/web-storage/offline-for-pwa#how_can_i_tell_how_much_storage_space_my_app_is_using) article, `Cache API falls into the “best effort” category which means the storage can be cleared without interrupting the user`, they don't provide more information about how it work.

If the Cache API is still not being used widely, a workaround for this maybe is to let the users know what repos that they're having the cache version and how many days have the cache been saved.